zipline/algorithm.py

```diff
@@ -1093,12 +1093,14 @@ def cancel_order(self, order_param):
 
     @api_method
     @require_initialized(HistoryInInitialize())
-    def history(self, sids, bar_count, frequency, field, ffill=True):
-        if self.data_portal is None:
-            raise Exception("no data portal!")
+    def history(self, assets, bar_count, frequency, field, ffill=True):
+        try:
```

Comment:
Just to be nice to users.  If they pass in a single asset, and it's painfully obvious what they were trying to do, I figured it would be nicer just to do what they wanted.

Furthermore, if they do pass in a single asset, I didn't have any checks verifying that.  So it was blowing up much deeper in dataportal, instead of earlier.  After adding the check, I figured I might as well coerce for them.